assert/assert.go

```diff
@@ -179,6 +175,15 @@ func logFailureFromBool(t TestingT, msgAndArgs ...interface{}) {
 	t.Log(format.WithCustomMessage(failureMessage+msg, msgAndArgs...))
 }
 
+func failureMsgFromError(err error) string {
+	// Handle errors with non-nil types
+	v := reflect.ValueOf(err)
+	if v.Kind() == reflect.Ptr && v.IsNil() {
+		return fmt.Sprintf("error is not nil: error has type %T", err)
```

Comment:
I still think that this could include the message if the call doesn't fail (https://github.com/gotestyourself/gotest.tools/issues/185#issuecomment-582745881), since calls on nil are valid so long as the method doesn't try and access its receiver as though it's not nil. A further improvement, though, and not strictly necessary to fix the other issues.

Reply:
That is true. I suspect that most `error` implementations which use a pointer receiver will require it to be non-nil. I would generally like to avoid the panic/recover, so would prefer to not add it for such a rare case, but it could be done as a future improvement.